### ISSUE TITLE
SC2032 triggers weirdly in a bigger file

### ISSUE THREAD

**Issue by stdedos** [koalaman/shellcheck#2353]
#### For bugs
- Rule Id (if any, e.g. SC1000): 
- My shellcheck version (`shellcheck --version` or "online"): 
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [ ] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [x] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

I have the following line in a (much bigger) file:
```sh
#!/bin/bash

alias rm='rm --one-file-system'
```

#### Here's what shellcheck currently says:

for which shellcheck says:
```
$ shellcheck myscript
No issues detected!

$
```

However, in a much bigger file (2.7k LOC), I get:

```
SC2032: Use own script or sh -c '..' to run this from su.
```

Which, I am not sure why it happens. There is no `su` in the file.

... after further investigation, there seems to be a line (further down into the file), that has a `... | xargs rm` pipeline.
I guess shellcheck's issue is that "it's afraid" that my newly-defined alias won't apply to my ^^pipeline?

**Comment by stdedos** [comment]
Related line is:
```bash
cdu -iBFsbz1024dh | grep -Po "\\.\\/.+" | xargs rm
```

which (correctly) triggers SC2033 online (instead of SC2032); however, I won't have access to the PC used until Monday.

It's weird, because I picked up https://github.com/koalaman/shellcheck/releases/tag/latest (exactly before reporting this bug), and I think it said the same 😕
However, the project is a tad bigger (there are no `see shellcheck -x` warnings, and I have activated the "stylistic" `${x}` enforce.

**Comment by stdedos** [comment]
> Can you find the line with the associated SC2033 and see what it is?

I am sorry - I am not sure what to further respond @koalaman. The facts are:

```bash
#!/bin/bash
alias rm='rm --one-file-system'
cdu -iBFsbz1024dh | grep -Po "\\.\\/.+" | xargs rm
```

give:

```
$ shellcheck myscript
 
Line 2:
alias rm='rm --one-file-system'
      ^-- SC2032 (info): Use own script or sh -c '..' to run this from xargs.
 
Line 3:
cdu -iBFsbz1024dh | grep -Po "\\.\\/.+" | xargs rm
                                                ^-- SC2033 (warning): Shell functions can't be passed to external commands.
```

I guess, I was focused on the "weird" "info" itself too much (
```
Line 2:
alias rm='rm --one-file-system'
      ^-- SC2032 (info): Use own script or sh -c '..' to run this from xargs.
```
) _instead of_ "ignoring" SC2032 - since it's "the other side of the SC2033-coin".

As "misleading" as the info line is, it's useful for pin-pointing "wte SC2033 is talking about", instead of having to guess "why is that SC2033 complaining?!!". If considered solely by itself, however, it is confusing to me.

I'd reword it as:
```
^-- SC2032 (info): Use own script or sh -c '..' to run this from xargs. (Caused by: SC2033, /path/to/file/x:123)
```

**Comment by koalaman** [comment]
I see what you mean. I updated the message to say "This function can't be invoked via xargs on line 3." (it's still not a function but hey). The suggestion to use a `sh -c` or a script is now on SC2033 instead, since that's where you'd want to do it. Thanks!
